lib/ruby_lsp/handler.rb

```diff
@@ -43,7 +43,7 @@ def on(msg, &blk)
 
     def handle(request)
       result = @handlers[request[:method]]&.call(request)
-      @writer.write(id: request[:id], result: result) if result
+      @writer.write(id: request[:id], result: result) unless result == false
```

Comment:
I think a better solution here (considering that LSP responses could return any type including `nil` and `false`) is:
1. Create a new constant named `Void = Object.new.freeze` that is used by requests to signal that they have no return value (basically what Sorbet runtime does with `void` returns)
2. Use that for requests that don't return a result.
3. Explicitly handle requests that we have no handlers for in this `handle` method.

Reply:
I think the approach would've worked. Returning `nil` for requests that aren't handled works and no request can return just `false`, because it's not JSON.

That said, I agree that the approach using `VOID` is more robust, so I switched.